sonar-docs/analysis/scan/sonarscanner-for-msbuild.md

```diff
@@ -0,0 +1,203 @@
+
+* [How to: Target a Version of the .NET Framework](https://msdn.microsoft.com/en-us/library/bb398202.aspx)
+* [MSBuild Target Framework and Target Platform](https://msdn.microsoft.com/en-us/library/hh264221.aspx)
+
+If you do not want to switch your production build to MSBuild 14.0, you can set up a separate build dedicated to the SonarQube analysis.
+
+**Detection of Test Projects**  
+SonarQube analyzes test projects differently from non-test projects, so it is important to correctly classify test projects.
+
+By default, the SonarQube Scanner for MSBuild will detect as test project:
+
+. MSTest unit test projects, thanks to the presence of a well-known project type GUID in .csproj file of such projects.
+1. Projects with names ending in "Test" or "Tests". This behavior can be changed by providing the parameter `sonar.msbuild.testProjectPattern` to the begin step (regex follows .NET Regular Expression in a case-sensitive way with the default value `.*Tests?\.(cs|vb)proj$`). This regex is applied against the fullname of the `.csproj` or `.vbproj` which is why it's recommended to keep at the end of your custom regex `\.(cs|vb)proj$`.
+To manually classify a project as a test project, mark it with `<SonarQubeTestProject>true</SonarQubeTestProject>`:
+```
+<!-- in .csproj -->
+<PropertyGroup>
+  <!-- Mark the project as being a test project -->
+  <SonarQubeTestProject>true</SonarQubeTestProject>
+</PropertyGroup>
+```
+
+**Per-project analysis parameters**
+Some analysis parameters can be set for a single MSBuild project by adding them to its .csproj file.
+```
+<!-- in .csproj -->
+<ItemGroup>
+  <SonarQubeSetting Include="sonar.stylecop.projectFilePath">
+    <Value>$(MSBuildProjectFullPath)</Value>
+  </SonarQubeSetting>
+</ItemGroup>
+```
+
+**Concurrent Analyses on the Same Build Machine**  
+Concurrent analyses (i.e. parallel analysis of two solutions on the same build machine using a unique service account) are not supported by default by the Scanner for MSBuild. You can enable it as follows:
+
+1. Locate the folder containing the Scanner for MSBuild
+1. Go in the `Targets` folder and copy the folder `SonarQube.Integration.ImportBefore.targets`
+1. Paste it under your build tool global `ImportBefore` folder (if the folder doesn't exist, create it).
+   * For MSBuild, the path is `<MSBUILD_INSTALL_DIR>\<Version>\Microsoft.Common.targets\ImportBefore` where <MSBUILD_INSTALL_DIR> is:
+      *For v14, default path is: `C:\Program Files (x86)\MSBuild\14.0\Microsoft.Common.Targets\ImportBefore`
+      * For v15, default path is: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Microsoft.Common.targets\ImportBefore` (for VS Community Edition)
+      * For v16, default path is: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Microsoft.Common.targets` (for VS Community Edition)
```

Comment:
Markdown error - the symbol before _C:\Program Files_ is an apostrophe, rather than a back-tick.